custom_components/tahoma/tahoma_api.py

```diff
@@ -629,19 +634,24 @@ def label(self):
 
     @property
     def command_definitions(self):
-        """List of command devinitions."""
-        return self.__definitions["commands"]
+        """List of command definitions."""
+        return self.__command_def
 
     @property
     def state_definitions(self):
-        """State of command devinition."""
-        return self.__definitions["states"]
+        """List of state definition."""
+        return self.__states_def
 
     @property
     def active_states(self):
         """Get active states."""
         return self.__active_states
 
+    @property
+    def attributes(self):
+        """Return entity attributes if any."""
+        return self.__attributes
```

Comment:
Why create such variable `self.__attributes`? You can directly use self.attributes without creating this property method.

Reply:
mostly because that's how the API was written by philklei. We do plan to rewrite the whole stuff anyways, so I wouldn't spend too much time cleaning it up for now.

Reply:
OK. Do you plan to contribute back to its repository or create your own?

Reply:
Not sure. We already made quite some changes, and in the end are planning to have an async api instead of the sync one we have ATM.

Reply:
@tetienne we are looking into rewriting the python api wrapper, where we start from scratch. If you are interested in contributing, let me know. We have an example in a separate repository. :-)

Reply:
For sure I'm interesting :) It sounds familiar to me :)

Reply:
I will add you later today, we are in the beginning phase, so would love to learn from your learning of the Somfy component.

We are now in a stage where we understand what functionality of the TaHoma API is needed. The current modified version in this repository doesn't have a high code quality, and I was not looking into improving the current one tbh.